api/src/main/java/com/datatorrent/api/DAG.java

```diff
@@ -288,4 +288,23 @@
   {
 
   }
+
+  /**
+   * This class provides functionality to extend existing DAG. The type of extensions allowed are
+   *
+   * <li>Add a new operator</li>
+   * <li>Remove an existing operator</li>
+   * <li>Add an operator to stream</li>
+   * <li>Add an stream between existing operator and new operator</li>
+   */
+  interface DAGChangeSet extends DAG
+  {
+    void removeOperator(String name);
+
+    void removeStream(String name);
+
+    StreamMeta extendStream(String id, Operator.InputPort... ports);
+
+    StreamMeta addStream(String id, String operatorName, String portName, Operator.InputPort... ports);
```

Comment:
As discussed on the call, it would be good to adopt a more object oriented and encapsulated approach. It will be good to handle the changes within the plan instead of passing around disjoined DAG fragments that require the use of String references. Is that doable?